table/tables/index.go

```diff
@@ -485,50 +498,89 @@ func (c *index) Exist(sc *stmtctx.StatementContext, txn kv.Transaction, indexedV
-		key, tempKey, keyVer = GenTempIdxKeyByState(c.idxInfo, key)
-		if keyVer != TempIndexKeyTypeNone {
-			KeyExistInfo, h1, err1 := KeyExistInTempIndex(context.TODO(), txn, tempKey, distinct, h, c.tblInfo.IsCommonHandle)
-			if err1 != nil {
-				return false, nil, err
-			}
-			switch KeyExistInfo {
-			case KeyInTempIndexNotExist, KeyInTempIndexIsDeleted:
-				return false, nil, nil
-			case KeyInTempIndexConflict:
-				return true, h1, kv.ErrKeyExists
-			case KeyInTempIndexIsItself:
-				continue
-			}
+		key, tempKey, _ := GenTempIdxKeyByState(c.idxInfo, key)
+		if len(tempKey) > 0 {
+			key = tempKey
 		}
-
-		value, err := txn.Get(context.TODO(), key)
-		if kv.IsErrNotFound(err) {
-			return false, nil, nil
+		foundKey, dupHandle, err := FetchDuplicatedHandle(context.TODO(), key, distinct, txn, c.tblInfo.ID, c.tblInfo.IsCommonHandle)
+		if err != nil || !foundKey {
+			return false, nil, err
 		}
-		if err != nil {
+		if dupHandle != nil && !dupHandle.Equal(h) {
 			return false, nil, err
 		}
+		continue
+	}
+	return true, h, nil
+}
 
-		// For distinct index, the value of key is handle.
-		if distinct {
-			var handle kv.Handle
-			handle, err := tablecodec.DecodeHandleInUniqueIndexValue(value, c.tblInfo.IsCommonHandle)
+// FetchDuplicatedHandle is used to find the duplicated row's handle for a given unique index key.
+func FetchDuplicatedHandle(ctx context.Context, key kv.Key, distinct bool,
+	txn kv.Transaction, tableID int64, isCommon bool) (foundKey bool, dupHandle kv.Handle, err error) {
+	val, err := txn.Get(ctx, key)
+	if err != nil {
+		if kv.IsErrNotFound(err) {
+			return false, nil, nil
+		}
+		// Unexpected errors.
+		return false, nil, err // Unexpected errors.
+	}
+	if isTemp, idxID := tablecodec.IsTempIndexKey(key); isTemp {
+		originValInTmp, deletedHandle, deleted, _, _ := tablecodec.DecodeTempIndexValue(val, isCommon)
+		if deleted {
+			originKey := key.Clone()
+			tablecodec.TempIndexKey2IndexKey(idxID, originKey)
+			originVal, err := txn.Get(ctx, originKey)
 			if err != nil {
+				if kv.IsErrNotFound(err) {
+					// The key has been deleted. This is not a duplicated key.
+					return false, nil, nil
+				}
+				// Unexpected errors.
 				return false, nil, err
 			}
-			if !handle.Equal(h) {
-				return true, handle, kv.ErrKeyExists
+			if !distinct {
+				return false, nil, nil
 			}
+			originHandle, err := tablecodec.DecodeHandleInUniqueIndexValue(originVal, isCommon)
+			if err != nil {
+				return false, nil, err
+			}
+			if originHandle.Equal(deletedHandle) {
+				// The key has been deleted. This is not a duplicated key.
+				return false, nil, nil
+			}
+			// The inequality means multiple modifications happened in the same key.
+			// We use the handle in origin index value to check if the row exists.
+			recPrefix := tablecodec.GenTableRecordPrefix(tableID)
+			rowKey := tablecodec.EncodeRecordKey(recPrefix, originHandle)
+			_, err = txn.Get(context.Background(), rowKey)
```

Comment:
Why use background()

Reply:
Otherwise we have to change the interface like `index.Create()`, and pass the context to a parameter like `CreateIdxOpt`. What's your suggestion?